payments/price_feed.py

```diff
@@ -1,47 +1,75 @@
 import requests
-
 import config
 
 
-def get_price(currency):
-    price_feed = "https://api.coindesk.com/v1/bpi/currentprice.json"
+def get_currency_provider_constants(currency=config.base_currency):
+    provider_settings = {
+        "COINDESK": {
+            "price_feed": "https://api.coindesk.com/v1/bpi/currentprice.json",
+            "result_root": "bpi",
+            "value_attribute": "rate_float",
+            "ticker": currency.upper()
+        },
+        "COINGECKO": {
+            "price_feed": "https://api.coingecko.com/api/v3/exchange_rates",
+            "result_root": "rates",
+            "value_attribute": "value",
+            "ticker": currency.lower()
+        }
+    }
+    return provider_settings
+
+
+def get_price(currency, currency_provider=config.currency_provider):
+
+    # Get currency_provider-specific settings
+    price_feed = get_currency_provider_constants(
+        currency)[currency_provider]["price_feed"]
+    result_root = get_currency_provider_constants(
+        currency)[currency_provider]["result_root"]
+    value_attribute = get_currency_provider_constants(
+        currency)[currency_provider]["value_attribute"]
+    ticker = get_currency_provider_constants(
+        currency)[currency_provider]["ticker"]
+
     r = requests.get(price_feed)
 
     for i in range(config.connection_attempts):
         try:
             price_data = r.json()
-            prices = price_data["bpi"]
+            prices = price_data[result_root]
             break
 
         except Exception as e:
             print(e)
             print(
-                "Attempting again... {}/{}...".format(i + 1, config.connection_attempts)
+                "Attempting again... {}/{}...".format(
+                    i + 1, config.connection_attempts)
             )
 
     else:
         raise ("Failed to reach {}.".format(price_feed))
 
     try:
-        price = prices[currency]["rate"].replace(",", "")
+        price = prices[ticker][value_attribute]
         return price
 
     except:
         print("Failed to find currency {} from {}.".format(currency, price_feed))
         return None
 
 
-def get_btc_value(dollar_value, currency):
+def get_btc_value(base_amount, currency):
     price = get_price(currency)
-    if price is not None:
 
+    if price is not None:
         try:
-            float_value = float(dollar_value) / float(price)
+            float_value = float(base_amount) / float(price)
             if not isinstance(float_value, float):
-                raise Exception("Dollar value should be a float.")
+                raise Exception("Fiat value should be a float.")
         except Exception as e:
             print(e)
 
         return float_value
```

Comment:
There is a potential bug in this code. (although I believe it is subsequently handled by later in the code, albeit not a little convoluted)

if price is 0 and `float_value = float(dollar_value) / float(price)` throws a `ZeroDivisionError` then the error will be printed out and the function will return `None` which will cause line 144 in `setSale.py` to error. It is good practice to handle the thrown from a function from the function calling it.

You may want to easily handle this by checking that price is not 0 and raising an exception if so.

Another good practice is to define your own exceptions for better handling. E.g.
```python
class FiatValueExpetion extends Exception:
    pass
```